server/apiRoutes/adminRoutes.js

```diff
@@ -0,0 +1,263 @@
+  }
+});
+
+/*
+ * @ROUTE: POST to /api/admin/extend_chore_time
+ * @DESC: Allows for an admin to extend a chores time by feeding new due date
+ * @ACCESS: admin only
+ */
+router.post('/extend_chore_time', async (req, res, next) => {
+  await updateChoreStatus();
+  const { userIdToModify, choreId, newDueDate } = req.body;
+  const adjusted = new Date(newDueDate);
+  console.log(adjusted);
+  AssignedChore.update(
+    { expiresOn: adjusted },
+    { where: { userId: userIdToModify, choreId } }
+  )
+    .then(newChore => res.send(newChore))
+    .catch(next);
+});
+
+/*
+ * @ROUTE: POST to /api/admin/add_new_user
+ * @DESC: Allows for an admin to extend a chores time
+ * @ACCESS: admin only
+ */
+router.post('/add_new_user', async (req, res, next) => {
+  await updateChoreStatus();
+  const { userId, firstName, surName, email, groupId, adminRights } = req.body;
+  let newUserId = 0;
+  const isAdmin = await UserGroup.findOne({ where: { userId, groupId } })
+    .then(user => {
+      if (user.userIsAdmin === true && user.userStatus === 'active') {
+        return true;
+      }
+      return false;
+    })
+    .catch(next);
+  if (isAdmin) {
+    const newUser = await User.create({ firstName, surName, email }).catch(
+      next
+    );
+    newUserId = newUser.id;
+    const group = await UserGroup.create({
+      userIsAdmin: adminRights,
+      userStatus: 'active',
+      userId: newUserId,
+      groupId: groupId,
+    })
+      .then(newUserGroup => res.send(`User Created Successfully`))
+      .catch(next);
+  }
+  res.send('You are not an Admin');
+});
+
+/*
+ * @ROUTE: POST to /api/admin/assign_chores
+ * @DESC: Route that will randomly assign all chores to the admins group
+ * @ACCESS: admin only
+ */
+router.post('/assign_chores', async (req, res, next) => {
+  const { userId } = req.body;
+  await updateChoreStatus();
+  const allChores = await Chore.findAll({});
+  const { userIsAdmin, groupId } = await UserGroup.findOne({
+    where: { userId },
+  });
+  let groupUsers = await UserGroup.findAll({ where: { groupId } });
+
+  //Slim out user Ids for the group Admin is in
+  let users = [];
+  for (let i = 0; i < groupUsers.length; i++) {
+    users.push(groupUsers[i].userId);
+  }
+
+  //Pull out all chores that are pending
+  const pendingChores = await AssignedChore.findAll({
+    where: { status: 'pending', userId: { [Op.in]: users } },
+  });
+  const pendingArr = [];
+  for (let i = 0; i < pendingChores.length; i++) {
+    pendingArr.push(pendingChores[i].id);
+  }
+  //console.log(pendingArr); need to exclude these
+
+  //slim out to only chores that need assigned
+  const choresToAssign = [];
+  for (let i = 0; i < allChores.length; i++) {
+    if (!pendingArr.includes(allChores[i].id)) {
+      choresToAssign.push(allChores[i].id);
+    }
+  }
+  //users and chores to assign
+  //stir it up ... little darling ... stir it up
+  shuffleArray(users);
+  shuffleArray(choresToAssign);
+
+  if (userIsAdmin) {
```

Comment:
you should check if user is admin at the very beginning all by itself
if(!userIsAdmin){
  return res.sendStatus(400)
}